sympy/matrices/normalforms.py

```diff
@@ -67,24 +71,27 @@ def invariant_factors(m, domain=None):
     return factors
 
 
-def hermite_normal_form(A, *, D=None, check_rank=False):
+def hermite_normal_form(A, mode, *, D=None, check_rank=False, truncate=True):
```

Comment:
Does this need to be added as a required argument without a default value? This is a backwards incompatible change.

Reply:
I think a kwarg with default value would be fine. Originally I felt it was important to force the user to be aware of the issue by making it a required arg, but perhaps that's overdoing it. (And not worth the compatibility break.)